Packs/TrendMicroVisionOne/Integrations/TrendMicroVisionOneEventCollector/TrendMicroVisionOneEventCollector.py

```diff
@@ -488,18 +405,14 @@ def get_datetime_range(
             settings={'TIMEZONE': 'UTC', 'RETURN_AS_TIMEZONE_AWARE': True}
         )
 
-    if log_type_time_field_name in (
-        LastRunLogsTimeFields.OBSERVED_ATTACK_TECHNIQUES.value, LastRunLogsTimeFields.SEARCH_DETECTIONS.value
-    ):
+    if log_type_time_field_name == LastRunLogsStartTimeFields.OBSERVED_ATTACK_TECHNIQUES.value:
+
         # Note: The data retrieval time range cannot be greater than 365 days for oat logs,
         # it cannot exceed datetime.now, otherwise the api will return 400
-        one_day_from_last_run_time = last_run_time_datetime + timedelta(  # type: ignore[operator]
-            hours=date_range_for_oat_and_search_logs  # type: ignore[operator]
+        one_year_from_last_run_time = last_run_time_datetime + timedelta(  # type: ignore[operator]
+            days=ONE_YEAR  # type: ignore[operator]
         )
-        if one_day_from_last_run_time > now:
-            end_time_datetime = now
-        else:
-            end_time_datetime = one_day_from_last_run_time
+        end_time_datetime = now if one_year_from_last_run_time > now else one_year_from_last_run_time
```

Comment:
whatre we doing here?

Reply:
the date retrieval time range cannot be greater than 365 days for oat logs,

now if the end_time for those logs are bigger than the datetime.now the api will return 400 Bad Request,

hence we need to make sure that in case we are one year from now and it exceeds the datetime.now, that we would just set the endtime to be datetime.now.